static/app/views/issueList/groupSearchViewTabs/issueViews.tsx

```diff
@@ -504,14 +496,59 @@ export function IssueViewsStateProvider({
     tempView: initialTempView,
   });
 
+  const debounceUpdateViews = useMemo(
+    () =>
+      debounce((newTabs: IssueView[]) => {
+        if (newTabs) {
+          updateViews({
+            orgSlug: organization.slug,
+            groupSearchViews: newTabs
+              .filter(tab => tab.isCommitted)
+              .map(tab => ({
+                // Do not send over an ID if it's a temporary or default tab so that
+                // the backend will save these and generate permanent Ids for them
+                ...(tab.id[0] !== '_' && !tab.id.startsWith('default')
+                  ? {id: tab.id}
+                  : {}),
+                name: tab.label,
+                query: tab.query,
+                querySort: tab.querySort,
+              })),
+          });
+        }
+      }, 500),
+    [organization.slug, updateViews]
+  );
+
+  const updateQueryParams = (params: IssueViewsQueryUpdateParams) => {
+    const {newQueryParams, replace = false} = params;
+    navigate(
+      normalizeUrl({
+        ...location,
+        query: {
+          ...queryParams,
+          ...newQueryParams,
+        },
+      }),
+      {replace}
+    );
+  };
+
   const dispatchWrapper = (action: IssueViewsActions) => {
     const newState = reducer(state, action);
     dispatch(action);
 
+    // These actions are outside of the dispatch to avoid introducing side effects in the reducer
     if (action.type === 'SYNC_VIEWS_TO_BACKEND' || action.syncViews) {
       debounceUpdateViews(newState.views);
     }
 
+    if (action.updateQueryParams) {
+      queueMicrotask(() => {
```

Comment:
Is this queueMicrotask necessary?

Reply:
This seemed to fix an error that would occur in tests (something about not being able to simultaneously update state in two components at once), though that never seemed to be an issue in the UI.